src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets

```diff
@@ -81,6 +81,12 @@
     <ProjectCapability Include="ProjectImportsTree" />
     <ProjectCapability Include="LaunchProfiles" />
     <ProjectCapability Include="NoGeneralDependentFileIcon"/>
+    <!--
+      We want the tree to respect IProjectTree2.DisplayOrder when it is set.
+      This is different to capability EditableDisplayOrder, used by F# for example, which
+      allows the user to manipulate display order values.
+    -->
+    <ProjectCapability Include="SortByDisplayOrder"/>
```

Comment:
This has a significant impact on performance of tree updates. This isn't designed to be enabled for large trees that you might see in C#.

Reply:
@lifengl is there a way we can enable this just for a graft and not the entire tree?

Reply:
I'm not convinced there's a significant perf impact. The only place CPS opts in to the capability is `DisplayOrderProjectScopeUnattachedProjectTreeNodeFactory` which derives from `ProjectScopeUnattachedProjectTreeNodeFactory` with higher precedence, customising two things:

1. Modifing the `IComparer<UnattachedProjectTreeNode>` used for sets of child nodes to be [`DisplayOrderProjectTreeSort`](http://index/?leftProject=Microsoft.VisualStudio.ProjectSystem.Implementation&file=Designers%5CDisplayOrderProjectTreeSort.cs) rather than `ProjectTreeSort`. This comparer's overhead is negligible.
1. Overriding [`IndexOf`](http://index/?leftProject=Microsoft.VisualStudio.ProjectSystem.Implementation&file=Designers%5CDisplayOrderProjectScopeUnattachedProjectTreeNodeFactory.cs&line=42) for looking up a child by caption. This override allocates less memory but has slightly worse algorithmic complexity than the base it replaces (O(N) vs O(4 log N)). I've filed https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/219245 which removes allocation by the override altogether.